TST, MAINT: test_theilslopes_warnings fails on 32-bit Windows

The status of the Windows builds in the wheels repo is now much better--down to a single failure for 32-bit Windows in https://github.com/MacPython/scipy-wheels/pull/166 (same for Python `3.8` and `3.9`; we don't currently support `3.10` with 32-bit):

```
================================== FAILURES ===================================
__________________________ test_theilslopes_warnings __________________________
[gw4] win32 -- Python 3.8.10 C:\Python38\python.exe
C:\Python38\lib\site-packages\scipy\stats\tests\test_mstats_basic.py:952: in test_theilslopes_warnings
    assert_allclose(res, (0, 0, np.nan, np.nan))
E   Failed: DID NOT WARN. No warnings of type (<class 'RuntimeWarning'>,) were emitted. The list of emitted warnings is: [].
        res        = TheilslopesResult(slope=0.0, intercept=0.0, low_slope=nan, high_slope=nan)
```

@mdhaber this was changed recently in gh-16056 -- maybe there's a somewhat simple path forward here? I can iterate a bit if you can guide me on the approach you'd like?

--- a/scipy/stats/tests/test_mstats_basic.py
+++ b/scipy/stats/tests/test_mstats_basic.py
@@ -947,7 +947,8 @@ def test_theilslopes_warnings():
     with pytest.warns(RuntimeWarning, match="All `x` coordinates are..."):
         res = mstats.theilslopes([0, 1], [0, 0])
         assert np.all(np.isnan(res))
-    with pytest.warns(RuntimeWarning, match="invalid value encountered..."):
+    with suppress_warnings() as sup:
+        sup.filter(RuntimeWarning, "invalid value encountered...")
         res = mstats.theilslopes([0, 0, 0], [0, 1, 0])
         assert_allclose(res, (0, 0, np.nan, np.nan))
 

